src/app-admin.js

```diff
@@ -0,0 +1,491 @@
+       FROM users
+       JOIN files ON files.owner = users.id
+       WHERE files.id = ?`,
+      [fileId, fileId],
+    );
+    accessMap.set(fileId, userAccess);
+  });
+
+  res.status(200).send({ status: 'ok', data: Array.from(accessMap.entries()) });
+});
+
+app.get('/access/check-access', (req, res) => {
+  const fileId = req.query.fileId;
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  const { id: fileIdInDb } = getFileById(fileId);
+  if (!fileIdInDb) {
+    sendErrorResponse(res, 400, 'invalid-file-id', 'File not found at server');
+    return false;
+  }
+
+  if (isAdmin(session.user_id)) {
+    res.json({ granted: true });
+    return;
+  }
+
+  const { owner } =
+    getAccountDb().first(
+      `SELECT files.owner
+     FROM files
+     WHERE files.id = ?`,
+      [fileId],
+    ) || {};
+
+  res.json({ granted: owner === session.user_id });
+});
+
+app.post('/access/transfer-ownership/', (req, res) => {
+  const newUserOwner = req.body || {};
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  if (!checkFilePermission(newUserOwner.fileId, session.user_id, res)) return;
+
+  if (!newUserOwner.newUserId) {
+    sendErrorResponse(res, 400, 'user-cant-be-empty', 'User cannot be empty');
+    return;
+  }
+
+  const { cnt } =
+    getAccountDb().first(
+      'SELECT count(*) AS cnt FROM users WHERE users.id = ?',
+      [newUserOwner.newUserId],
+    ) || {};
+
+  if (cnt === 0) {
+    sendErrorResponse(res, 400, 'new-user-not-found', 'New user not found');
+    return;
+  }
+
+  getAccountDb().mutate('UPDATE files SET owner = ? WHERE id = ?', [
+    newUserOwner.newUserId,
+    newUserOwner.fileId,
+  ]);
+
+  res.status(200).send({ status: 'ok', data: {} });
+});
+
+app.get('/file/owner', async (req, res) => {
+  const fileId = req.query.fileId;
+  const session = validateUser(req, res);
+  if (!session) return;
+
+  if (!checkFilePermission(fileId, session.user_id, res)) return;
+
+  let canGetOwner = await isAdmin(session.user_id);
+  if (!canGetOwner) {
+    const { canListAvaiableUserFromDB } =
+      getAccountDb().first(
+        `SELECT count(*) as canListAvaiableUserFromDB
+       FROM files
+       WHERE files.id = ? and files.owner = ?`,
+        [fileId, session.user_id],
+      ) || {};
+    canGetOwner = canListAvaiableUserFromDB === 1;
+  }
+
+  if (canGetOwner) {
+    const owner =
+      getAccountDb().first(
+        `SELECT users.id, users.user_name userName, users.display_name as displayName
+     FROM files
+      JOIN users
+      ON users.id = files.owner
+     WHERE files.id = ?`,
+        [fileId],
+      ) || {};
+
+    res.json(owner);
+  }
+
+  return null;
+});
+
+app.get('/auth-mode', (req, res) => {
+  const { method } =
+    getAccountDb().first(
+      `SELECT method from auth
+        where active = 1`,
+    ) || {};
+
+  res.json({ method });
+});
+
+app.get('/multiuser', (req, res) => {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I wonder if we could return the state of multi-user as part of the `bootstrap` endpoint. Or perhaps one of the auth endpoints.

Reply:
bump

Reply:
moved it to "/needs-bootstrap"